### PR TITLE
Added cmdlets and tests or the .NET SDK version 4.0.1

### PR DESCRIPTION
**Pull request by prakharsharma10**
## Description

This PR has been created because there were a lot of merge conflicts with the upstream branch in this PR https://github.com/Azure/azure-powershell/pull/5703 .  All the comments have already been addressed.

The changes in this PR include
1. Added PS models for 'delivery policy' which has been added as a property on a Azure CDN endpoint
2. Added a cmdlet validating probe URL
3. Added cmdlets for enabling and disabling custom HTTPS
4. Added cmdlet for getting supported optimization types for a CDN profile
5. Added tests for all the new cmdlets

Link for .Net SDK https://github.com/Azure/azure-sdk-for-net/tree/psSdkJson6/src/SDKs/Cdn 

Cmdlet review here: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/23

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Cdn/Commands.Cdn.Test/Commands.Cdn.Test.csproj [hunk 1]
```diff
@@ -78,6 +77,14 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.3\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn.Test/Commands.Cdn.Test.csproj]: @prakharsharma10 you can remove the references to `Microsoft.Rest.ClientRuntime` and Microsoft.Rest.ClientRuntime.Azure` in this file since they are pulled in from a separate targets file

File: src/ResourceManager/Cdn/Commands.Cdn.Test/packages.config [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
   <package id="Microsoft.Azure.Management.Resources" version="2.20.0-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Test.Framework" version="1.0.6179.26854-prerelease" targetFramework="net45" />
-  <package id="Microsoft.Azure.Test.HttpRecorder" version="1.6.7-preview"  targetFramework="net45" />
+  <package id="Microsoft.Azure.Test.HttpRecorder" version="1.6.7-preview" targetFramework="net45" />
   <package id="Microsoft.Bcl" version="1.1.9" targetFramework="net45" />
   <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net45" />
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.10" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn.Test/packages.config]: @prakharsharma10 same comment about removing these two references as they are pulled in elsewhere

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomain.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        [Parameter(Mandatory = true, HelpMessage = "The resource group of the Azure CDN profile.", ParameterSetName = FieldsParameterSet)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The custom domain object.", ParameterSetName = ObjectParameterSet)]
+        [ValidateNotNull]
+        public PSCustomDomain InputObject { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Return object (if specified).")]
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomain.cs]: @prakharsharma10 please remove the `ValueFromPipelineByPropertyName` attribute from this parameter

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomain.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        [Parameter(Mandatory = true, HelpMessage = "The resource group of the Azure CDN profile.", ParameterSetName = FieldsParameterSet)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The custom domain object.", ParameterSetName = ObjectParameterSet)]
+        [ValidateNotNull]
+        public PSCustomDomain InputObject { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Return object (if specified).")]
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomain.cs]: @prakharsharma10 please remove the `ValueFromPipelineByPropertyName` property from this parameter

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomain.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+using Microsoft.Azure.Commands.Cdn.Models.CustomDomain;
+using Microsoft.Azure.Commands.Cdn.Properties;
+using Microsoft.Azure.Management.Cdn;
+using System.Linq;
+using Microsoft.Azure.Commands.Cdn.Models.Endpoint;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Cdn.CustomDomain
+{
+    [Cmdlet(VerbsLifecycle.Disable, "AzureRmCdnCustomDomain", DefaultParameterSetName = FieldsParameterSet, SupportsShouldProcess = true), OutputType(typeof(bool))]
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomain.cs]: @prakharsharma10 this cmdlet says it implements `ShouldProcess`, but there is no `ShouldProcess` method call wrapping the cmdlet execution below, so if a user provides `-WhatIf` to see what will happen when they run this command, it will actually run the command, which isn't the experience we want. Check out [this document](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/should-process-confirm-impact.md) for more information on using this method.

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomain.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+using Microsoft.Azure.Commands.Cdn.Models.CustomDomain;
+using Microsoft.Azure.Commands.Cdn.Properties;
+using Microsoft.Azure.Management.Cdn;
+using System.Linq;
+using Microsoft.Azure.Commands.Cdn.Models.Endpoint;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Cdn.CustomDomain
+{
+    [Cmdlet(VerbsLifecycle.Enable, "AzureRmCdnCustomDomain", DefaultParameterSetName = FieldsParameterSet, SupportsShouldProcess = true), OutputType(typeof(bool))]
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/EnableAzureRmCdnCustomDomain.cs]: @prakharsharma10 same comment about implementing `ShouldProcess`

File: src/ResourceManager/Cdn/Commands.Cdn/Profile/GetAzureRmCdnProfileSupportedOptimizationType.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            if (ParameterSetName == ObjectParameterSet)
+            {
+                ResourceGroupName = CdnProfile.ResourceGroupName;
+                ProfileName = CdnProfile.Name;
+            }
+            var supportedOptimizationTypes =
+                CdnManagementClient.Profiles.ListSupportedOptimizationTypes(ResourceGroupName, ProfileName).SupportedOptimizationTypes.Select(optimizationType => new PSOptimizationType(optimizationType)).ToList();
+
+            WriteVerbose(Resources.Success);
+            WriteObject(supportedOptimizationTypes);
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/Profile/GetAzureRmCdnProfileSupportedOptimizationType.cs]: @prakharsharma10 this should be `WriteObject(supportedOptimizationTypes, true)`

File: src/ResourceManager/Cdn/Commands.Cdn/help/Get-AzureRmCdnEndpointNameAvailability.md [hunk 8]
```diff
@@ -65,9 +65,6 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 
 ## INPUTS
 
-### None
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/help/Get-AzureRmCdnEndpointNameAvailability.md]: @prakharsharma10 please revert this change (even if it was made by `platyPS`)

File: src/ResourceManager/Cdn/Commands.Cdn/help/Get-AzureRMCdnProfile.md [hunk 9]
```diff
@@ -80,9 +80,6 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 
 ## INPUTS
 
-### None
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/help/Get-AzureRMCdnProfile.md]: @prakharsharma10 please revert this change (even if it was made by `platyPS`)

File: src/ResourceManager/Cdn/Commands.Cdn.Test/Commands.Cdn.Test.csproj (hunk 1 above)
Comment (prakharsharma10) [reply src/ResourceManager/Cdn/Commands.Cdn.Test/Commands.Cdn.Test.csproj]: @cormacpayne We are using clientruntime version 2.3.10 instead of 2.3.11 Therefore, we need to have these references.

File: src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomain.cs (hunk 5 above)
Comment (prakharsharma10) [reply src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/DisableAzureRmCdnCustomDomain.cs]: @cormacpayne Added ConfirmAction method as done for other cmdlets in CDN that supports shouldProcess

**Comment by prakharsharma10** [comment]
@cormacpayne I have addressed the comments. Please take a look. Thanks!

File: src/ResourceManager/Cdn/Commands.Cdn.Test/Commands.Cdn.Test.csproj (hunk 1 above)
Comment (cormacpayne) [reply src/ResourceManager/Cdn/Commands.Cdn.Test/Commands.Cdn.Test.csproj]: @prakharsharma10 the SDK has a minimum version dependency on 2.3.10, so it is able to use 2.3.11. Deleting the added reference will use 2.3.11 (and 3.3.12 of the other package), which is what we want

**Comment by cormacpayne** [comment]
@prakharsharma10 as mentioned in the comment about, management SDKs have a minimum version dependency on the ClientRuntime, which means you will be able to use 2.3.11 (and 3.3.12 of the other package), which is the version targeted in our [`Common.Dependencies.Test.targets`](https://github.com/Azure/azure-powershell/blob/preview/tools/Common.Dependencies.Test.targets) file. These are the versions of the ClientRuntime packages that we ship, so we need to be running tests against those specific versions to ensure there are no breaks.

**Comment by prakharsharma10** [comment]
@cormacpayne All comments addressed. Clientruntime issue fixed. Please take a look

File: src/ResourceManager/Cdn/Commands.Cdn/help/Confirm-AzureRmCdnEndpointProbeURL.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+## DESCRIPTION
+The **Confirm-AzureRmCdnEndpointProbeURL** cmdlet confirms if the probe URL provided can be used for dynamic site acceleration.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> Confirm-AzureRmCdnEndpointProbeURL -ProbeUrl "http://www.bing.com/images"
```
Comment (cormacpayne) [inline src/ResourceManager/Cdn/Commands.Cdn/help/Confirm-AzureRmCdnEndpointProbeURL.md]: @prakharsharma10 for the cmdlets you are adding in this PR, would you mind updating the examples for their markdown files to include the output of the command that is run? That will give users a better idea of what is returned from the cmdlets

Comment (prakharsharma10) [reply src/ResourceManager/Cdn/Commands.Cdn/help/Confirm-AzureRmCdnEndpointProbeURL.md]: @cormacpayne We don't have the examples for many of the cmdlets and as we are working on a release, it's not a high priority for us right now. Some of the customers have already requested for these cmdlets and would appreciate if we could publish these cmdlets ASAP and add the examples later. Thanks!

Comment (markcowl) [reply src/ResourceManager/Cdn/Commands.Cdn/help/Confirm-AzureRmCdnEndpointProbeURL.md]: @prakharsharma10 This is a requirement for changes in the stable branch.  Please make this change.

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/756/
